vllm/model_executor/models/blip2.py

```diff
@@ -707,8 +707,7 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
             use_default_weight_loading = False
             if "vision" in name:
                 if self.vision_model is not None:
-                    # We only do sharding for language model and
-                    # not vision model for now.
+                    # BlipVisionModel does not need sharding
```

Comment:
Oh, this is because the original implementation also uses the `qkv` for the attention mechanism.
As far as I understand, the sharding is for mapping the checkpoints `q_proj`, `k_proj`, and `v_proj` for the vLLM model's `qkv_proj`.
However, the BlipVisionModel also uses `qkv` projection, so it does not need mapping.
Do I understand correctly?

Reply:
Yep, I forgot to update this comment but I went back to the `transformers` implementation and that was indeed the case. Thanks for clarifying!